src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/BalanceSheetServiceImpl.java

```diff
@@ -72,66 +83,152 @@ public BalanceSheet getBalanceSheet(String transactionId, String companyAccounts
+                    throw new ServiceException("Bad request posting balance sheet", e);
+                }
+            }
         } catch (URIValidationException e) {
-
             throw new ServiceException("Invalid URI for current period resource", e);
         }
-
-        return new ArrayList<>();
     }
 
     @Override
     public BalanceSheetHeadings getBalanceSheetHeadings(CompanyProfileApi companyProfile) {
+        boolean isSameYear = isSameYear(companyProfile);
+        BalanceSheetHeadings balanceSheetHeadings = new BalanceSheetHeadings();
+        balanceSheetHeadings.setPreviousPeriodHeading(getPreviousPeriodHeading(companyProfile, isSameYear));
+        balanceSheetHeadings.setCurrentPeriodHeading(getCurrentPeriodHeading(companyProfile, isSameYear));
+        return balanceSheetHeadings;
+    }
 
-        LocalDate currentPeriodEndOn =
-                convertDateTimeToLocalDate(companyProfile.getAccounts().getNextAccounts().getPeriodEndOn());
-        LocalDate currentPeriodStartOn =
-                convertDateTimeToLocalDate(companyProfile.getAccounts().getNextAccounts().getPeriodStartOn());
+    private String getCurrentPeriodHeading(CompanyProfileApi companyProfile, boolean isSameYear) {
+        NextAccountsApi nextAccountsApi = companyProfile.getAccounts().getNextAccounts();
+        LocalDate currentPeriodEndOn = convertDateTimeToLocalDate(nextAccountsApi.getPeriodEndOn());
+        LocalDate currentPeriodStartOn = convertDateTimeToLocalDate(nextAccountsApi.getPeriodStartOn());
 
-        boolean isSameYear = false;
+        return accountsDatesHelper.generateBalanceSheetHeading(currentPeriodStartOn, currentPeriodEndOn, isSameYear);
+    }
 
-        if (companyProfile.getAccounts().getLastAccounts() != null &&
-                companyProfile.getAccounts().getLastAccounts().getPeriodEndOn() != null) {
+    private String getPreviousPeriodHeading(CompanyProfileApi companyProfile, boolean isSameYear) {
+        LastAccountsApi lastAccountsApi = companyProfile.getAccounts().getLastAccounts();
+        if (isMultipleYearFiler(companyProfile)) {
+            LocalDate previousPeriodStartOn = convertDateTimeToLocalDate(lastAccountsApi.getPeriodStartOn());
+            LocalDate previousPeriodEndOn = convertDateTimeToLocalDate(lastAccountsApi.getPeriodEndOn());
 
-            LocalDate previousPeriodEndOn =
-                convertDateTimeToLocalDate(companyProfile.getAccounts().getLastAccounts().getPeriodEndOn());
-            isSameYear = accountsDatesHelper.isSameYear(previousPeriodEndOn, currentPeriodEndOn);
+            return accountsDatesHelper.generateBalanceSheetHeading(previousPeriodStartOn, previousPeriodEndOn, isSameYear);
         }
+        return null;
+    }
 
-        String currentPeriodHeading = accountsDatesHelper.generateBalanceSheetHeading(currentPeriodStartOn, currentPeriodEndOn, isSameYear);
+    private boolean isSameYear(CompanyProfileApi companyProfile) {
```

Comment:
The `isSameYear()` method would be better named `isSameYearFiler()` to add additional context, and for consistency with the `isMultipleYearFiler()` method.